### PR TITLE
Add actions support

### PR DESCRIPTION
**Pull request by zhang-hua**
# Description and Motivation 

...

# Change type 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

---
This change is [Reviewable](https://reviewable.io/reviews/azure/pcs-cli/380)

### CODE REVIEW

File: src/deploymentmanager.ts [hunk 1]
```diff
@@ -523,17 +530,16 @@ export class DeploymentManager implements IDeploymentManager {
...
-        data.push(`PCS_IOTHUBREACT_HUB_ENDPOINT="Endpoint=${outputs.eventHubEndpoint.value}"`);
-        data.push(`PCS_IOTHUBREACT_HUB_PARTITIONS=${outputs.eventHubPartitions.value}`);
-        data.push(`PCS_IOTHUBREACT_HUB_NAME=${outputs.eventHubName.value}`);
-        data.push(`PCS_IOTHUBREACT_AZUREBLOB_ACCOUNT=${outputs.storageAccountName.value}`);
-        data.push(`PCS_IOTHUBREACT_AZUREBLOB_KEY="${outputs.storageAccountKey.value}"`);
-        data.push(`PCS_IOTHUBREACT_AZUREBLOB_ENDPOINT_SUFFIX=${storageEndpointSuffix}`);
         data.push(`PCS_ASA_DATA_AZUREBLOB_ACCOUNT=${outputs.storageAccountName.value}`);
         data.push(`PCS_ASA_DATA_AZUREBLOB_KEY="${outputs.storageAccountKey.value}"`);
         data.push(`PCS_ASA_DATA_AZUREBLOB_ENDPOINT_SUFFIX=${storageEndpointSuffix}`);
+        data.push(`PCS_AZUREBLOB_ACCOUNT=${outputs.storageAccountName.value}`);
```
Comment (molly-moen) [inline src/deploymentmanager.ts]: are these three variables (blob account, key, endpoint) used anywhere currently? If they aren't and we are just going to use them for telemetry, we should change them into 1 variable, the connection string. The format is AccountName={AccountName};AccountKey={AccountKey};EndpointSuffix={EndpointSuffix}

Comment (zhang-hua) [reply src/deploymentmanager.ts]: I think they are used in ASA manager: https://github.com/Azure/remote-monitoring-services-dotnet/blob/master/asa-manager/WebService/appsettings.ini#L44.

Comment (molly-moen) [reply src/deploymentmanager.ts]: yes the asa_data ones are, but if we replace the iothubreact_ ones with a single connection string variable, we can transition asa to also use the connection string later, then delete the asa_data ones, which will save us 3 environment variables

File: solutions/remotemonitoring/single-vm/setup.sh [hunk 2]
```diff
@@ -202,18 +205,17 @@ echo "export PCS_STORAGEADAPTER_DOCUMENTDB_CONNSTRING=\"${PCS_STORAGEADAPTER_DOC
...
-echo "export PCS_IOTHUBREACT_HUB_PARTITIONS=\"${PCS_IOTHUBREACT_HUB_PARTITIONS}\""                       >> ${ENVVARS}
-echo "export PCS_IOTHUBREACT_AZUREBLOB_ACCOUNT=\"${PCS_IOTHUBREACT_AZUREBLOB_ACCOUNT}\""                 >> ${ENVVARS}
-echo "export PCS_IOTHUBREACT_AZUREBLOB_KEY=\"${PCS_IOTHUBREACT_AZUREBLOB_KEY}\""                         >> ${ENVVARS}
-echo "export PCS_IOTHUBREACT_AZUREBLOB_ENDPOINT_SUFFIX=\"${PCS_IOTHUBREACT_AZUREBLOB_ENDPOINT_SUFFIX}\"" >> ${ENVVARS}
-echo "export PCS_ASA_DATA_AZUREBLOB_ACCOUNT=\"${PCS_IOTHUBREACT_AZUREBLOB_ACCOUNT}\""                    >> ${ENVVARS}
-echo "export PCS_ASA_DATA_AZUREBLOB_KEY=\"${PCS_IOTHUBREACT_AZUREBLOB_KEY}\""                            >> ${ENVVARS}
-echo "export PCS_ASA_DATA_AZUREBLOB_ENDPOINT_SUFFIX=\"${PCS_IOTHUBREACT_AZUREBLOB_ENDPOINT_SUFFIX}\""    >> ${ENVVARS}
+echo "export PCS_AZUREBLOB_ACCOUNT=\"${PCS_AZUREBLOB_ACCOUNT}\""                                         >> ${ENVVARS}
+echo "export PCS_AZUREBLOB_KEY=\"${PCS_AZUREBLOB_KEY}\""                                                 >> ${ENVVARS}
+echo "export PCS_AZUREBLOB_ENDPOINT_SUFFIX=\"${PCS_AZUREBLOB_ENDPOINT_SUFFIX}\""                         >> ${ENVVARS}
```
Comment (molly-moen) [inline solutions/remotemonitoring/single-vm/setup.sh]: do we need to export the 3 components of the connection string now that we have the connection string as one var?

File: solutions/remotemonitoring/single-vm/setup.sh [hunk 3]
```diff
@@ -45,14 +44,21 @@ while [ "$#" -gt 0 ]; do
         --aad-instance)                 PCS_WEBUI_AUTH_AAD_INSTANCE="$2" ;;
         --release-version)              PCS_RELEASE_VERSION="$2" ;;
         --docker-tag)                   PCS_DOCKER_TAG="$2" ;;
-        --evenhub-connstring)           PCS_EVENTHUB_CONNSTRING="$2" ;;
+        --eventhub-connstring)          PCS_EVENTHUB_CONNSTRING="$2" ;;
```
Comment (ppathan) [inline solutions/remotemonitoring/single-vm/setup.sh]: I didn't check but his typo is fixed in arm template as well?

File: solutions/remotemonitoring/single-vm/setup.sh [hunk 4]
```diff
@@ -194,26 +200,19 @@ echo "export HOST_NAME=\"${HOST_NAME}\""
 echo "export APP_RUNTIME=\"${APP_RUNTIME}\""                                                             >> ${ENVVARS}
 echo "export PCS_AUTH_ISSUER=\"${PCS_AUTH_ISSUER}\""                                                     >> ${ENVVARS}
 echo "export PCS_AUTH_AUDIENCE=\"${PCS_AUTH_AUDIENCE}\""                                                 >> ${ENVVARS}
-echo "export PCS_AUTH_AAD_GLOBAL_TENANTID=\"${PCS_AUTH_AAD_GLOBAL_TENANTID}\""                           >> ${ENVVARS}
-echo "export PCS_AUTH_AAD_GLOBAL_CLIENTID=\"${PCS_AUTH_AAD_GLOBAL_CLIENTID}\""                           >> ${ENVVARS}
-echo "export PCS_AUTH_AAD_GLOBAL_LOGINURI=\"${PCS_AUTH_AAD_GLOBAL_LOGINURI}\""                           >> ${ENVVARS}
```
Comment (ppathan) [inline solutions/remotemonitoring/single-vm/setup.sh]: Do we not need this?

File: src/azurehelper.ts [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    private SLEEP_TIME = 5000;
+
+    constructor(environment: AzureEnvironment, subscriptionId: string, credentials: ServiceClientCredentials) {
+        this._environment = environment;
+        this._subscriptionId = subscriptionId;
+        this._credentials = credentials;
+    }
+
+    public assignContributorRoleOnSubscription(principalId: string): Promise<boolean> {
+        const roleId = 'b24988ac-6180-42a0-ab88-20f7382dd24c';
```
Comment (ppathan) [inline src/azurehelper.ts]: Add a comment that this GUID represents roleId as `Contributor` in AAD. Also make this a static variable and call it CONTRIBUTOR_ID

File: src/azurehelper.ts [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+        const scope = `/subscriptions/${this._subscriptionId}`;
+        return this.createRoleAssignmentWithRetry(principalId, scope, roleId);
+    }
+
+    public assignContributorRoleOnResourceGroup(principalId: string, resourceGroupName: string): Promise<boolean> {
+        const roleId = 'b24988ac-6180-42a0-ab88-20f7382dd24c';
```
Comment (ppathan) [inline src/azurehelper.ts]: Add a comment that this GUID represents roleId as `Contributor` in AAD. Also make this a static variable and call it CONTRIBUTOR_ID

File: src/azurehelper.ts [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+        const scope = `/subscriptions/${this._subscriptionId}/resourceGroups/${resourceGroupName}`;
+        return this.createRoleAssignmentWithRetry(principalId, scope, roleId);
+    }
+
+    public assignOwnerRoleOnSubscription(principalId: string): Promise<boolean> {
+        const roleId = '8e3af657-a8ff-443c-a75c-2fe8c4bcb635';
```
Comment (ppathan) [inline src/azurehelper.ts]: ditto comment as before but for `Owner`

File: src/azurehelper.ts [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+        const scope = `/subscriptions/${this._subscriptionId}`;
+        return this.createRoleAssignmentWithRetry(principalId, scope, roleId);
+    }
+
+    public assignOwnerRoleOnResourceGroup(principalId: string, resourceGroupName: string): Promise<boolean> {
+        const roleId = '8e3af657-a8ff-443c-a75c-2fe8c4bcb635';
```
Comment (ppathan) [inline src/azurehelper.ts]: ditto comment as before but for `Owner`

File: src/index.ts [hunk 9]
```diff
@@ -584,9 +585,15 @@ function createServicePrincipal(azureWebsiteName: string,
         return createAppRoleAssignment(adminAppRoleId, sp, graphClient, baseUri);
     })
     .then((sp: any) => {
+        options.tokenAudience = undefined;
```
Comment (ppathan) [inline src/index.ts]: Wondering if we can move the entire createServicePrincipal in the AzureHelper. BTW, I like the clean separate with AzureHelper!!

File: solutions/remotemonitoring/single-vm/setup.sh (hunk 3 above)
Comment (zhang-hua) [reply solutions/remotemonitoring/single-vm/setup.sh]: yes.

File: solutions/remotemonitoring/single-vm/setup.sh (hunk 4 above)
Comment (zhang-hua) [reply solutions/remotemonitoring/single-vm/setup.sh]: yes, I did not see any place to use it after searching it in java and dotnet code.

File: src/azurehelper.ts (hunk 5 above)
Comment (zhang-hua) [reply src/azurehelper.ts]: yep. This is the only place to use this id. Encapsulate it in the function won't expose it to other place to use it directly. Actually the Guid is not reference by anywhere.

**Pull request merged**